provider/pkg/provider/suite_test.go

```diff
@@ -277,41 +216,13 @@ func (c *providerTestContext) NewProvider(opts ...NewProviderOption) *kubeProvid
 	Expect(err).ShouldNot(HaveOccurred())
 
 	k.makeClient = func(ctx context.Context, config *rest.Config) (*clients.DynamicClientSet, *clients.LogClient, error) {
-		// make a fake clientset for testing purposes, backed by an testing.ObjectTracker with pre-populated objects.
-		// see also: https://github.com/kubernetes/client-go/blob/kubernetes-1.29.0/examples/fake-client/main_test.go
-		disco := &discoveryfake.FakeDiscovery{
-			Fake:               &kubetesting.Fake{},
-			FakedServerVersion: &options.ServerVersion,
-		}
-		mapper := meta.NewDefaultRESTMapper([]schema.GroupVersion{})
-		client := dynamicfake.NewSimpleDynamicClient(scheme.Scheme, options.Objects...)
-		cs := &clients.DynamicClientSet{
-			GenericClient:         client,
-			DiscoveryClientCached: &mockCachedDiscoveryClient{DiscoveryInterface: disco},
-			RESTMapper:            &mockRestMapper{RESTMapper: mapper},
-		}
-
-		// make a fake log client for testing purposes.
-		clientset := fake.NewSimpleClientset(options.Objects...)
-		lc := clients.NewLogClient(ctx, clientset.CoreV1())
-
-		return cs, lc, nil
+		dc, _, _, _ := fakeclients.NewSimpleDynamicClient(options.copts...)
```

Comment:
nit: add a simple comment why we ignore errs.

Reply:
Those aren't errors, they're the underlying fake clients.